### DISCUSSION TITLE
ask about switch option

### DISCUSSION THREAD

**Discussion by wezo3** [savonet/liquidsoap#2719] [General]
Hello,
About switch option in liquidsoap I need to start as one time only this supported with switch or not ?
For example I need to switch to start playlist or event as one time from specific time to other specific time 
like this start at 1 November 10:50 to 2 November 11:30

how can do this with liquidsoap and if this still not available can please review this feature to add soon in next releases ?

Also I thought to start this by my script but there issue must be kill old liquidsoap process and start it again that effect to stream performance to resume previous playlist, so I need send to start one time at the time and end at the time and fallback again to previous playlist

Regards

**Comment by wezo3** [comment]
Any hint about this, guys ?

**Reply by toots** [reply]
Hi @wezo3. The most straight forward way would be to write a (fast) function (it can be an external script) that returns `true` during the time you want the source to play. Then you can make the source available only during that time and for its play using a switch. Typically:

```
def should_the_source_be_available() =
  # Returns true or false
   ....
end

source_to_make_available = ...

s = switch(track_sensitive=false, [
  (should_the_source_be_available, source_to_make_available),
  # Add your other sources here..
])
```

**Reply by toots** [reply]
`time.local` returns a record with methods for day, month, year, hour, minute etc. You can definitely use it to implement your function:
```
# time.local();;
- :
  {
    dst = false,
    year_day = 330,
    week_day = 6,
    year = 2022,
    month = 11,
    day = 26,
    hour = 9,
    min = 37,
    sec = 50
  }
```
